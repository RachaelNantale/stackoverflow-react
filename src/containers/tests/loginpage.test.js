import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import LoginPageConnected, { mapDispatchToProps, LoginPage } from '../login';
import ACTION_TYPE from '../../actions/actionTypes';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('<LoginPage />', () => {
  let AuthPageComponent;
  let wrapper;
  beforeEach(() => {
    const initialState = { loginReducer: { freshUser: { email: '', password: '' } } };
    const store = mockStore(initialState);
    AuthPageComponent = mount(<Provider store={store}><LoginPageConnected /></Provider>);
  });

  it('should render the component', () => {
    expect(AuthPageComponent).toMatchSnapshot();
  });

  it('should dispatch a method to get  login user input', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).getUserInputs({});
    expect(dispatch.mock.calls[0][0]).toEqual({ payload: {}, type: ACTION_TYPE.GET_USER_INPUT });
    mapDispatchToProps(dispatch).loginUser({});
  });

  it('should call handle input user method', () => {
    const getUserInputs = jest.fn();
    wrapper = shallow(
      <LoginPage
        loginUser={jest.fn()}
        freshUser={{}}
        getUserInputs={getUserInputs}
      />,
    );
    wrapper.instance().handleUpdateFields({ target: { name: 'username', value: 'rachael' } });
    expect(getUserInputs).toHaveBeenCalled();
  });
});


describe('handle login', () => {
  const wrapper = shallow(<LoginPage />);
  wrapper.setProps({ loginUser: jest.fn() });
  const fakeEventReturn = { target: { id: 1, value: 'some val' } };
  const fakeEvent = { preventDefault: () => fakeEventReturn };
  const spy = jest.spyOn(wrapper.instance(), 'handleLogin');
  wrapper.instance().handleLogin(fakeEvent);
  expect(spy).toHaveBeenCalled();
});
