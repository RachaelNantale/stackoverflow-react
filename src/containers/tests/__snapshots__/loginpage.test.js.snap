// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> should render the component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(LoginPage)>
    <LoginPage
      freshUser={
        Object {
          "email": "",
          "password": "",
        }
      }
      getUserInputs={[Function]}
      loginUser={[Function]}
    >
      <Signup
        onChange={[Function]}
        onSubmit={[Function]}
        title="Login"
      >
        <div
          className=" signup-container"
        >
          <form
            className="centered-form"
            onSubmit={[Function]}
          >
            <h3
              className="text-center "
            >
              Login
            </h3>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="email"
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
              />
            </div>
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </Signup>
    </LoginPage>
  </Connect(LoginPage)>
</Provider>
`;
